### PR TITLE
Add an initial StepAction for ephemeral cluster provisioning

### PR DESCRIPTION
**Pull request by amisstea**

### CODE REVIEW

**Comment by amisstea** [comment]
@Omeramsc please review. @lcarva I believe you've been looking at StepActions recently so it'd be great to get your eyes on this too.

File: stepactions/get-supported-ephemeral-cluster-versions/0.1/README.md
Comment (Omeramsc) [inline stepactions/get-supported-ephemeral-cluster-versions/0.1/README.md:9]: I wonder if there are any risks using skipping the verification. if there are, we might want to mention them

Comment (amisstea) [reply stepactions/get-supported-ephemeral-cluster-versions/0.1/README.md]: Added a warning.

**Review by Omeramsc — APPROVED** [review]
LGTM

**Comment by amisstea** [comment]
Moving to draft state. I'm planning to make changes to the params based on a recent decision to make the EaaS subspaces ephemeral.

**Comment by amisstea** [comment]
@Omeramsc please take another look at the latest changes. I'll open a separate PR for creation of the ephemeral EaaS space.

**Review by Omeramsc — APPROVED** [review]
LGTM

**Comment by amisstea** [comment]
@konflux-ci/build-maintainers can I please get some feedback (and possibly an approval) on this PR?

**Comment by chmeliik** [comment]
> @konflux-ci/build-maintainers can I please get some feedback (and possibly an approval) on this PR?

I don't have any context, but I can check the Tekton-isms and give the green checkmark based on approval from your team

Though I would appreciate some basic context for why this is a StepAction and who/what will use it

File: stepactions/eaas-get-supported-ephemeral-cluster-versions/0.1/eaas-get-supported-ephemeral-cluster-versions.yaml
Comment (chmeliik) [inline stepactions/eaas-get-supported-ephemeral-cluster-versions/0.1/eaas-get-supported-ephemeral-cluster-versions.yaml:43]: It would be good to be careful with the quoting, doubly so because INSECURE_SKIP_TLS_VERIFY is user-specified
```suggestion
    OC=(oc --insecure-skip-tls-verify="$INSECURE_SKIP_TLS_VERIFY")
    SV=$("${OC[@]}" get configmap supported-versions -n hypershift -o=jsonpath='{.data.supported-versions}')
```

File: stepactions/eaas-get-supported-ephemeral-cluster-versions/0.1/eaas-get-supported-ephemeral-cluster-versions.yaml
Comment (chmeliik) [inline stepactions/eaas-get-supported-ephemeral-cluster-versions/0.1/eaas-get-supported-ephemeral-cluster-versions.yaml:44]: nitpick: you can avoid the `echo` 
```suggestion
    VERSIONS=$(jq -c '.versions' <<< "$SV")
```

File: stepactions/eaas-get-supported-ephemeral-cluster-versions/0.1/eaas-get-supported-ephemeral-cluster-versions.yaml
Comment (amisstea) [reply stepactions/eaas-get-supported-ephemeral-cluster-versions/0.1/eaas-get-supported-ephemeral-cluster-versions.yaml]: Good catch. Fixed.

File: stepactions/eaas-get-supported-ephemeral-cluster-versions/0.1/eaas-get-supported-ephemeral-cluster-versions.yaml
Comment (amisstea) [reply stepactions/eaas-get-supported-ephemeral-cluster-versions/0.1/eaas-get-supported-ephemeral-cluster-versions.yaml]: Updated.

**Comment by amisstea** [comment]
> Though I would appreciate some basic context for why this is a StepAction and who/what will use it

This StepAction is meant to be used with a few others (see #1067) for Konflux EaaS (Environment as a Service). They're a set of utilities which can be combined (as needed, or with custom steps) to provision ephemeral clusters from a pipelinerun. Use of StepActions, in particular, is a workaround (see [here](https://github.com/redhat-appstudio/infra-deployments/pull/3693#issuecomment-2108081165) for the explanation). This StepAction _could_ be a task but I think this way aligns with the [motivations](https://github.com/tektoncd/community/blob/main/teps/0142-enable-step-reusability.md#motivation) behind StepActions.

**Comment by chmeliik** [comment]
The tests seem to be failing on 
```
timed out when waiting for devfile content creation for application ...
```

That may be fixed by rebasing now that https://github.com/konflux-ci/build-definitions/pull/1066 is in

**Comment by amisstea** [comment]
@chmeliik please merge this if you can.

**Pull request merged**
